lib/phoenix_live_view/test/upload_client.ex

```diff
@@ -84,7 +79,38 @@ defmodule Phoenix.LiveViewTest.UploadClient do
     end
   end
 
-  defp build_and_join_entry(%{socket: nil} = _state, client_entry, token) do
+  defp build_join_entries(state, entries, entries_resp) do
+    Enum.reduce_while(entries, {:ok, %{}}, fn client_entry, {:ok, acc} ->
+      %{
+        "ref" => ref,
+        "name" => name,
+        "content" => content,
+        "size" => client_size
+      } = client_entry
+
+      token = Map.fetch!(entries_resp, ref)
+      content_size = byte_size(content)
+
+      if content_size == client_size do
+        {:cont,
+         {:ok, Map.put(acc, name, build_and_join_entry(state, client_entry, content_size, token))}}
+      else
+        reason = """
```

Comment:
can we simply raise here or is the issue this is running outside of the caller?

Reply:
correct, but on second thought let's not do this at all 😂